keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1083,6 +1105,38 @@ repeated evaluations if they are shared across validations.
     - expression: "itemMetadataNames.exists(name, name == 'required')"
 ```
 
+Since the variable names must be valid CEL names, this is checked at creation
+time.
+
+For resource constraint purposes, lazily-evaluated variables
+count towards the overall policy limit, but only once, to account for
+memoization.
```

Comment:
Specify that this is runtime cost?  (CEL admission does not have estimated cost)

What does only once mean? Does it mean that the runtime cost of a variable is computed when it is first evaluated and then added to the runtime cost of all expressions when evaluated? When exactly during evaluation does this cost get tallied?  This impacts when evaluation gets canceled due to the runtime cost limit being exceeded.

Please put answer in KEP.

Reply:
Gotcha, I've made things a bit more clear (explicitly referring to runtime cost, for instance). I think I mentioned to you offline the other day about making variables count towards the runtime cost limit every time they're referenced, though I decided against it since I think 1) while a bit more difficult, it's still simple enough for users to reason about runtime cost 2) I think it will help in cases where expensive computations are left to a variable.